tests/raftstore/test_stale_peer.rs

```diff
@@ -151,13 +163,18 @@ fn test_stale_peer_without_data<T: Simulator>(cluster: &mut Cluster<T>) {
     // Sleep one more second to make sure there is enough time for the peer to be destroyed.
     thread::sleep(Duration::from_secs(1));
 
+    // There must be no data on store 2 belongs to new region
+    must_get_none(&engine2, b"k3");
+
     // Check whether peer(2, 2) is destroyed.
     // Before peer 2 is destroyed, a tombstone mark will be written into the engine.
     // So we could check the tombstone mark to make sure peer 2 is destroyed.
-    let engine = cluster.get_engine(2);
-    let state_key = keys::region_state_key(1);
-    let state: RegionLocalState = engine.get_msg(&state_key).unwrap().unwrap();
+    let state_key = keys::region_state_key(new_region_id);
+    let state: RegionLocalState = engine2.get_msg(&state_key).unwrap().unwrap();
     assert_eq!(state.get_state(), PeerState::Tombstone);
+
+    // other region should not be affected.
```

Comment:
Please update the comments for the whole test case, so it's easier for the source code reader to understand why a region split is involved and the destroy should not delete other region's data of the same store.